### PR TITLE
SequenceIndexOp for saturate_index and wrap_index

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

**Comment by youknowone** [comment]
@falsetru @killme2008 Could you please review this PR as recent users of this functions?

**Review by killme2008 — APPROVED** [review]
LGTM

File: vm/src/sliceable.rs [hunk 1]
```diff
@@ -291,50 +292,59 @@ impl SequenceIndex {
...
+            p = len;
+        }
+        p as usize
     }
-    if p > len {
-        p = len;
+
+    fn wrapped_at(&self, len: usize) -> Option<usize> {
+        let neg = self.is_negative();
+        let p = self.wrapping_abs() as usize;
```
Comment (falsetru) [inline vm/src/sliceable.rs:320]:
```suggestion
        let p = self.unsigned_abs();
```

**Review by falsetru — APPROVED** [review]
Looks good. suggested minor changes.

File: vm/src/sliceable.rs [hunk 2]
```diff
@@ -291,50 +292,49 @@ impl SequenceIndex {
...
-    let mut p = p;
-    if p < 0 {
-        p += len;
-        if p < 0 {
-            p = 0;
-        }
+impl SequenceIndexOp for isize {
+    fn saturated_at(&self, len: usize) -> usize {
+        let len = len.to_isize().unwrap_or(Self::MAX);
+        *self.clamp(&0, &len) as usize
```
Comment (falsetru) [inline vm/src/sliceable.rs:305]: I didn't mean to remove `if p < 0` completely.

```python
let mut p = *self;
if p < 0 {
    p += len;
}
```

Something like :point_up: should be done before clamp() to handle `p <  0 and 0 <= p + len < len` case.

**Review by falsetru — APPROVED** [review]
LGTM

**Comment by youknowone** [comment]
Thank you!

**Pull request merged**